site/app/models/DisplayImage.php

```diff
@@ -0,0 +1,212 @@
+     * form factor.  If the larger dimension is already smaller than $max_dimension then don't do anything.
+     *
+     * @param \Imagick $image The image to resize
+     * @param int $max_dimension Length in pixels to resize the larger dimension to
+     * @throws \ImagickException
+     */
+    public static function resizeMaxDimension(\Imagick &$image, int $max_dimension): void {
+        $cols = $image->getImageWidth();
+        $rows = $image->getImageHeight();
+
+        if ($cols >= $rows && $cols > $max_dimension) {
+            $image->scaleImage($max_dimension, 0);
+        }
+        elseif ($rows > $max_dimension) {
+            $image->scaleImage(0, $max_dimension);
+        }
+    }
+
+    /**
+     * Save a new DisplayImage image into the file directory
+     *
+     * @param Core $core The application core
+     * @param string $user_id The user_id who will own this image
+     * @param string $new_image_name Name of the file to be saved, without the extension.  For example 'aphacker'
+     * @param string $image_extension File extension, for example 'jpeg' or 'gif'
+     * @param string $tmp_file_path Path to the temporary location of the file to work with.  This may be the temporary
+     *                              path in the $_FILES array or the location of a file after unzipping a zip archive
+     * @param string $folder The folder to save to, for example 'system_images' or 'user_images'
+     * @throws BadArgumentException $folder is not a member of self::LEGAL_FOLDERS
+     * @throws FileWriteException Unable to write to the image directory
+     * @throws \ImagickException
+     */
+    public static function saveUserImage(Core $core, string $user_id, string $new_image_name, string $image_extension, string $tmp_file_path, string $folder): void {
+        // Validate folder
+        if (!in_array($folder, self::LEGAL_FOLDERS)) {
+            throw new BadArgumentException('The $folder parameter must be a member of DisplayImage::LEGAL_FOLDERS.');
+        }
+
+        // Generate the path to the folder where this image should be saved
+        $folder_path = FileUtils::joinPaths(
+            $core->getConfig()->getSubmittyPath(),
+            'user_data',
+            $user_id,
+            $folder
+        );
+
+        // Generate the folder if it does not exist
+        if (!FileUtils::createDir($folder_path, true)) {
+            throw new FileWriteException('Error creating the user\'s system images folder.');
+        }
+
+        // Decrease image size while maintaining form factor
+        // If bigger image dimension is already smaller then IMG_MAX_DIMENSION don't do any resizing.
+        $imagick = new \Imagick($tmp_file_path);
+        self::resizeMaxDimension($imagick, self::IMG_MAX_DIMENSION);
+
+        // Save file (will overwrite any image with same name)
+        $imagick->writeImage(FileUtils::joinPaths($folder_path, $new_image_name . '.' . $image_extension));
+    }
```

Comment:
Neither of these functions make any usage of the model code other than a class constant. Please move them into a `ImageUtil` class under `app/libraries` similar to the other static functions we have in this vein.

Reply:
I'm going to shelve this until a later PR.  See #5587.